Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerImageAttachmentsView.swift

```diff
@@ -5,44 +5,39 @@
 import StreamChat
 import UIKit
 
+/// A view that displays a collection of image attachments
 public typealias ChatMessageComposerImageAttachmentsView = _ChatMessageComposerImageAttachmentsView<NoExtraData>
 
+/// A view that displays a collection of image attachments
 open class _ChatMessageComposerImageAttachmentsView<ExtraData: ExtraDataTypes>: _View,
     UIConfigProvider,
     UICollectionViewDelegate,
     UICollectionViewDataSource {
-    // MARK: - Properties
-    
-    open var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
-    
-    public var images: [UIImage] = [] {
+    /// The image attachment item size. Can be overridden in `setUpLayout()`.
+    private var imagePreviewItemSize: CGSize = .init(width: 100, height: 100)
+
+    /// The images data source.
+    public var content: [UIImage] = [] {
         didSet {
             updateContentIfNeeded()
         }
```

Comment:
It seems like this should be abstracted somehow. Rather than `UIImage` we should probably introduce some kind of protocol that any attachment can conform to. This will be extremely important for custom attachment types. 

Maybe something like this? We could also add more options to specify the source of the preview. Later we can extend it by `previewTitle` `previewSize` etc.
```
protocol PreviewProvider { 
  var previewImageURL: URL? { get }
  var previewImage: UIImage? { get }
}
```

Reply:
We also have this on the documents attachment view: 
![image](https://user-images.githubusercontent.com/12814114/114011340-051b7c80-985d-11eb-9324-a3aefe3ccf5a.png)
So we have some other "preview types", maybe just calling this `ImagePreviewProvider` ? Also, if the customer wants to add his own type, he would need to cast to his type to use other properties and fill them in the updateContent right?  🤔